frontends/web/src/routes/device/bitbox02/setup/name.tsx

```diff
@@ -38,10 +41,22 @@ export const SetDeviceName = ({
 }: TSetDeviceNameProps) => {
   const { t } = useTranslation();
   const [deviceName, setDeviceName] = useState('');
+  const [error, setError] = useState<'too-long' | 'invalid-chars' | false>();
 
   const handleDeviceNameInput = (event: Event) => {
     const target = (event.target as HTMLInputElement);
     const value: string = target.value;
+    const valueTrimmed: string = value.trim();
+
+    if (valueTrimmed.length < 1) {
+      setError(undefined);
+    } else if (valueTrimmed.length > 30) {
+      setError('too-long');
+    } else if (!regexPrintableAsciiOrSpace.test(valueTrimmed)) {
+      setError('invalid-chars');
+    } else {
+      setError(false);
```

Comment:
`return` after setting the error? calling `setDeviceName(...)` will fail anyway.

Reply:
well need deviceName always to be the current value. As it is later always passed back to the Input field no matter if it is valid or not.

https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable

Reply:
ah i misread and thought this invokes the set device name thing. nvm :+1: